### ISSUE TITLE
Improve NO_COLOR value set/test documentation

### ISSUE THREAD

**Issue by condekind** [jcs/no_color#83]
**tldr;** NO_COLOR is meant to be changed/tested. Make sure the (i) values to be set and (ii) common/intuitive tests are foolproof.

Considering this is the only mention of the variable's value:

> All command-line software which outputs text with ANSI color added should check for the presence of a NO_COLOR environment variable that, when present (regardless of its value), prevents the addition of ANSI color.

There are two problems I'm addressing in this issue, both related to POSIX compliant shells behavior:
**(1)**. Unsetting a variable or setting it to empty are different things.
**(2)**. Testing for an unset variable is [often done in the wrong way](https://stackoverflow.com/a/13864829/6587731)

Just to be clear: there's nothing strictly wrong with the way the standard was defined. This is simply a suggestion that takes into account the tricky/weird/old/esoteric nature of (POSIX compliant) shell scripts. Knowing that NO_COLOR is used mostly/solely for command line applications, it would make sense to shape it in a way that a straighforward use with basic shell constructs won't expose some tricky shell corner cases.

Examples of possibly unintended behavior:

```sh
# Error 1: The user wants colors and decides to "erase the variable's value".
# User mistakenly sets NO_COLOR to empty string instead of unsetting it:
export NO_COLOR=
sh goodProgram.sh
...
# The correct check is performed in goodProgram.sh, but NO_COLOR is set, so colors are not displayed:
if [ -z ${NO_COLOR+x} ]; then
  /usr/bin/ls --color=always
elif
  /usr/bin/ls --color=never
fi
# Conclusion: wrong result - the user made a mistake (1)

# Error 2: The user doesn't want colors anymore and decides to set the variable.
# User learned with past mistakes that empty string counts as set, and proudly sets it as:
export NO_COLOR=
sh badProgram.sh
...
# An erroneous check is performed in badProgram.sh and colors are displayed:
if [ -z "$NO_COLOR" ]; then
  /usr/bin/ls --color=always
elif
  /usr/bin/ls --color=never
fi
# Conclusion: wrong result - the developer of badProgram.sh made a mistake (2)

# Error 3 - the real bug shady: both user and developer make the mistakes above.
# User wants colors and thinks this will unset NO_COLOR:
export NO_COLOR=
sh badProgram.sh
...
# Developer thinks this will test if NO_COLOR is set:
if [ -z "$NO_COLOR" ]; then
  /usr/bin/ls --color=always
elif
  /usr/bin/ls --color=never
fi
# Conclusion: correct result. Both user and developer go on with their lives without realizing their mistakes.
```

The suggestions I came up with (which are not necessarily good):
If changing the standard is on the table:
* Instead of accepting any set value as valid, consider allowing only a single string. Devs would need to test NO_COLOR against a fixed value, such as "true". This covers the unset/set-to-empty/set-to-word seen above, but exposes the ["if $var; then..."](https://stackoverflow.com/a/21210966/6587731) mistake that could happen if the expected value were "true", 1, etc. Although this last one could be avoided with the use of a word such as "on" or "set", it's a rookie mistake, I wouldn't be so concerned. Besides, we'd be shifting a need for *intermediate* shell knowledge from both users and devs to a need for *basic* shell usage from the devs only (which are expected to know better).

If changing the standard is unlikely or impossible:
* Write explicit suggestions on the project's landing page, so that:
  (i) users set the value to a non-zero-length word such as "on", "set" or "true" (optionally emphasize that (1)); and
  (ii) developers test it properly. A short **GOOD/BAD** code example showing a test mistake would educate devs *and* serve as a hot Control+C target.

Note:
* I avoided the use of [[ tests for maximum portability (POSIX sh).
* I am aware of #79, and although I agree that reference implementations in other languages would be nice, I'm being specific to POSIX sh, and I do not address the need of a "YES_COLOR".

**Comment by jcs** [comment]
The spirit of the spec is to check if the environment variable is set and contains anything ("1", "true", "on", "off", "maybe", whatever) and if so, consider it enabled.  If it's in the environment but an empty value, it should be considered not set since it's likely to be assumed that the user meant to unset it.  Anything that `[ -z "$NO_COLOR" ]` returns true on should activate it.  I specifically worded it with "regardless of its value" to avoid any ambiguity over what the value should be.

**Comment by jcs** [comment]
I've updated the wording of the spec to explicitly state that the value of the variable should not be empty.
